finder-app/writer.c

```diff
@@ -0,0 +1,81 @@
+/*ECEN-5713 Advanced Embedded Software Development
+Author - Khyati Satta
+Date - 26 January 2023
+File Description - The following application is to write a given expression to a file using File IO operations
+*/
+
+#include<stdio.h>
+#include<syslog.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include<string.h>
+
+#define ERROR (-1u)
+
+//Function to write the given expression to the file
+int fileWrite(int file_descriptor , char *file_name , char *expr)
+{
+    syslog(LOG_DEBUG , "Writing %s to %s ", expr , file_name);
+
+    // Write to the file using write system call
+    int error_code = write(file_descriptor , expr , strlen(expr));
+
+    // Error check 1: Check if the write operation was successful
+    // It will return -1 if the write operation was interrupted
+    if(error_code == ERROR){
+        syslog(LOG_ERR , "The write operation for the string %s to the file %s was unsucessful!\n" , expr , file_name);
+        return 1;
+    }
+    // Error check 2: Check if all the bytes were written
+    else if(error_code != strlen(expr)){
+        syslog(LOG_ERR , "All the bytes were not written to the file. Only %d were written\n" , error_code);
```

Comment:
@KhyatiSatta Shouldn't the file be closed in this path? `close(file_descriptor);`